app/services/merge_requests/update_service.rb

```diff
@@ -23,11 +23,15 @@ def execute(merge_request)
         merge_request.update_nth_task(params[:task_num].to_i, false)
       end
 
+      old_labels = merge_request.labels.map{ |label| label }
```

Comment:
What about `merge_request.labels.to_a`? That will remove the reference to the actual collection and it won't be changed.